websites/M/MEE6/presence.ts

```diff
@@ -2,59 +2,202 @@ const presence = new Presence({
-	};
+			largeImageKey: Assets.Logo,
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname, hostname, href } = document.location,
+		[newLang, privacy, buttons, covers] = await Promise.all([
+			presence.getSetting<string>("lang").catch(() => "en"),
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("buttons"),
+			presence.getSetting<boolean>("covers"),
+		]),
+		pathnameSplit = pathname.split("/");
 
-	if (document.location.hostname === "mee6.xyz") {
-		if (document.location.pathname.includes("/leaderboard/")) {
-			title = document.querySelector<HTMLElement>(
-				"#app-mount > div > div > div > div.leaderboardHeader > div.leaderboardHeaderGuildInfo > div.leaderboardGuildName"
-			);
-			presenceData.details = "Viewing leaderboard of server:";
-			presenceData.state = title.textContent;
-		} else if (
-			document.querySelector(
-				"#app-mount > div > div.sc-e2wlkd-0.ideomx > div.sc-e2wlkd-2.bnyPVx > div > div > div.sc-13ru8pb-14.bhtyFP > h4"
-			)
-		) {
-			title = document.querySelector(
-				"#app-mount > div > div.sc-e2wlkd-0.ideomx > div.sc-e2wlkd-2.bnyPVx > div > div > div.sc-13ru8pb-14.bhtyFP > h4"
+	if (oldLang !== newLang || !strings) {
+		oldLang = newLang;
+		strings = await getStrings();
+	}
+	if (privacy) {
+		presenceData.details = strings.browse;
+		presence.setActivity(presenceData);
+		return;
+	}
+	switch (hostname.replace("www.", "")) {
+		case "mee6.xyz": {
+			if (document.querySelector('[class="sc-hKMtZM bHyJDw"]')) {
+				presenceData.details = `Rank card | ${
+					document.querySelector('[class="sc-jgbSNz bsZrWF"]')?.textContent
+				}`;
+				presenceData.largeImageKey =
+					document.querySelector("svg > image")?.getAttribute("xlink:href") ??
+					Assets.Logo;
+			} else {
+				switch (pathnameSplit[2]) {
+					case "": {
+						presenceData.details = strings.viewHome;
+						break;
+					}
+					case "plugins": {
+						presenceData.details = `Viewing plugin ${pathnameSplit[3]}`;
+						break;
+					}
+					case "monetize": {
+						presenceData.details = "Viewing monetisation options";
+						break;
+					}
+					case "custom-bot": {
+						presenceData.details = "Viewing custom bot options";
+						break;
+					}
+					case "pro":
+					case "premium": {
+						presenceData.details = "Viewing premium & pro options";
+						break;
+					}
+					case "careers": {
+						presenceData.details = "Viewing open careere positions";
+						break;
+					}
+					case "terms": {
+						presenceData.details = "Reading the terms";
+						presenceData.smallImageKey = Assets.Reading;
+						break;
+					}
+					case "privacy": {
+						presenceData.details = "Reading the privacy statement";
+						presenceData.smallImageKey = Assets.Reading;
+						break;
+					}
+					case "memberships": {
+						presenceData.details = "Managing their subscriptions";
+						break;
+					}
+					case "dashboard": {
+						switch (pathnameSplit[4]) {
+							case "": {
+								if (pathnameSplit[3] !== "") {
```

Comment:
Empty string is falsy 
```suggestion
								if (pathnameSplit[3]) {
```